src/org/omegat/core/data/ParseEntry.java

```diff
@@ -133,8 +140,9 @@ public void linkPrevNextSegments() {
      *            protected parts
      */
     @Override
-    public void addEntryWithProperties(String id, String source, String translation, boolean isFuzzy, String[] props,
-            String path, IFilter filter, List<ProtectedPart> protectedParts) {
+    public void addEntryWithProperties(String id, String source, String translation, boolean isFuzzy,
+            String[] props, String path, IFilter filter, List<ProtectedPart> protectedParts,
+            boolean isFinal) {
```

Comment:
One remark : I would use isLocked instead of isFinal (in ParseEntry, IParseCallback and SourceTextEntry). These classes are shared by all file formats, and some file formats may want to lock some segments for any other reason than a final status.